### ISSUE TITLE
Get-LatestFlash is not returning latest update across all OS

### ISSUE THREAD

**Issue by ah-uk** [aaronparker/LatestUpdate#32]
# Expected Behavior

Get-LatestFlash returns latest available update along with any required non superseded updates.

# Current Behavior

Get-LatestFlash is only returning update for October 2018 across all OS. I think that KB4462930 is not included in the monthly package so I think that is it probably correct to be offered but the latest monthly package which appears to be almost cumulative in all but the title is not being offered.

Atom feed suggests information is present to be returned here is one example..

```  
<entry>
    <title>Security Update for Adobe Flash Player: March 12, 2019</title>
    <id>urn:docid:4489907</id>
    <updated>2019-03-12T17:10:10Z</updated>
    <content type="text">Fixes vulnerabilities in Adobe Flash Player on certain versions of Windows. </content>
    <link href="https://support.microsoft.com/help/4489907" />
  </entry>
```

# Failure Information (for bugs)
```
PS C:\Windows\system32> Get-LatestFlash -Verbose
VERBOSE: Downloading feed of updates https://support.microsoft.com/app/content/api/content/feeds/sap/en-us/6ae59d69-36fc-8e4d-23dd-631d98bf74a9/atom.
VERBOSE: GET https://support.microsoft.com/app/content/api/content/feeds/sap/en-us/6ae59d69-36fc-8e4d-23dd-631d98bf74a9/atom with 0-byte payload
VERBOSE: received -1-byte response of content type application/atom+xml; charset=utf-8
VERBOSE: Reading RSS XML from C:\Users\ASHLEY~1.ADM\AppData\Local\Temp\1nqtpd3b.1ez.
VERBOSE: Deleting C:\Users\ASHLEY~1.ADM\AppData\Local\Temp\1nqtpd3b.1ez.
VERBOSE: Searching http://www.catalog.update.microsoft.com/Search.aspx?q=KB4462930.
VERBOSE: GET http://www.catalog.update.microsoft.com/Search.aspx?q=KB4462930 with 0-byte payload
VERBOSE: received 105259-byte response of content type text/html; charset=utf-8
VERBOSE: Contructing temporary table with KB, ID and URL.
VERBOSE: Checking Microsoft Update Catalog for Id: a15bd597-1c77-467c-8880-4502d754cdf4.
VERBOSE: POST http://www.catalog.update.microsoft.com/DownloadDialog.aspx with -1-byte payload
VERBOSE: received 32173-byte response of content type text/html; charset=utf-8
VERBOSE: Adding http://download.windowsupdate.com/d/msdownload/update/software/updt/2018/10/windows10.0-kb4462930-x64_aa56f192a75aac0029cd9a39d5ebf3eb625feebd.msu to output.
VERBOSE: Checking Microsoft Update Catalog for Id: 219441c8-ee87-47d2-865a-07bde7870989.
VERBOSE: POST http://www.catalog.update.microsoft.com/DownloadDialog.aspx with -1-byte payload
VERBOSE: received 32180-byte response of content type text/html; charset=utf-8
VERBOSE: Adding http://download.windowsupdate.com/d/msdownload/update/software/updt/2018/10/windows10.0-kb4462930-arm64_4331c3089be4dfd515059818f8b6c015c5643ad0.msu to output.
VERBOSE: Checking Microsoft Update Catalog for Id: f1cdbdf2-91ba-416e-bccf-8cfabdcecc8a.
VERBOSE: POST http://www.catalog.update.microsoft.com/DownloadDialog.aspx with -1-byte payload
VERBOSE: received 32102-byte response of content type text/html; charset=utf-8
VERBOSE: Adding http://download.windowsupdate.com/d/msdownload/update/software/updt/2018/10/windows10.0-kb4462930-x86_ee44cbbd86b3271990ccc472760f0228094131e5.msu to output.
VERBOSE: Checking Microsoft Update Catalog for Id: 3b80c9f0-1416-4f38-bc39-e68ec91b71fa.
VERBOSE: POST http://www.catalog.update.microsoft.com/DownloadDialog.aspx with -1-byte payload
VERBOSE: received 32172-byte response of content type text/html; charset=utf-8
VERBOSE: Adding http://download.windowsupdate.com/d/msdownload/update/software/updt/2018/10/windows10.0-kb4462930-x64_b22236a150c0fffaa1b132c7dfbbc844ad6c45e3.msu to output.
VERBOSE: Checking Microsoft Update Catalog for Id: b8ea1a5b-3636-4905-ae4b-d171014df7a6.
VERBOSE: POST http://www.catalog.update.microsoft.com/DownloadDialog.aspx with -1-byte payload
VERBOSE: received 32207-byte response of content type text/html; charset=utf-8
VERBOSE: Adding http://download.windowsupdate.com/d/msdownload/update/software/updt/2018/10/windows10.0-kb4462930-x64_b22236a150c0fffaa1b132c7dfbbc844ad6c45e3.msu to output.
VERBOSE: Checking Microsoft Update Catalog for Id: eecd3528-0fdc-4296-b164-71d4f4786d2e.
VERBOSE: POST http://www.catalog.update.microsoft.com/DownloadDialog.aspx with -1-byte payload
VERBOSE: received 32182-byte response of content type text/html; charset=utf-8
VERBOSE: Adding http://download.windowsupdate.com/c/msdownload/update/software/updt/2018/10/windows10.0-kb4462930-arm64_332ecfb5958a87eb6e37206cbd9cf1aa2339659e.msu to output.
VERBOSE: Checking Microsoft Update Catalog for Id: b2ef7350-9a3c-4471-9523-cb7c9b9be21b.
VERBOSE: POST http://www.catalog.update.microsoft.com/DownloadDialog.aspx with -1-byte payload
VERBOSE: received 32180-byte response of content type text/html; charset=utf-8
VERBOSE: Adding http://download.windowsupdate.com/c/msdownload/update/software/updt/2018/10/windows10.0-kb4462930-arm64_24b09da230c6a260882df62d211e6d1a26ad5dc0.msu to output.
VERBOSE: Checking Microsoft Update Catalog for Id: de111e67-04cb-40b6-811e-c0fec6aa5041.
VERBOSE: POST http://www.catalog.update.microsoft.com/DownloadDialog.aspx with -1-byte payload
VERBOSE: received 32102-byte response of content type text/html; charset=utf-8
VERBOSE: Adding http://download.windowsupdate.com/c/msdownload/update/software/updt/2018/10/windows10.0-kb4462930-x86_f3fb8219e526d228e9ae9b44973d7d73376f3367.msu to output.
VERBOSE: Checking Microsoft Update Catalog for Id: fe76c944-eeea-43f7-8b29-bb74d10a34cb.
VERBOSE: POST http://www.catalog.update.microsoft.com/DownloadDialog.aspx with -1-byte payload
VERBOSE: received 32113-byte response of content type text/html; charset=utf-8
VERBOSE: Adding http://download.windowsupdate.com/d/msdownload/update/software/updt/2018/10/windows10.0-kb4462930-x86_72c899f691403365053da2c53d1097f5065da3c7.msu to output.
VERBOSE: Checking Microsoft Update Catalog for Id: c32fc63b-de7a-49e3-9129-ae543d281446.
VERBOSE: POST http://www.catalog.update.microsoft.com/DownloadDialog.aspx with -1-byte payload
VERBOSE: received 32102-byte response of content type text/html; charset=utf-8
VERBOSE: Adding http://download.windowsupdate.com/c/msdownload/update/software/updt/2018/10/windows10.0-kb4462930-x86_b1bd42a524bb4abbd08f66090b69f9432a756de2.msu to output.
VERBOSE: Checking Microsoft Update Catalog for Id: 97ad809a-0df1-432c-b16c-948bc610b632.
VERBOSE: POST http://www.catalog.update.microsoft.com/DownloadDialog.aspx with -1-byte payload
VERBOSE: received 32172-byte response of content type text/html; charset=utf-8
VERBOSE: Adding http://download.windowsupdate.com/c/msdownload/update/software/updt/2018/10/windows10.0-kb4462930-x64_3a4a1cada2d518585873b6cfdad8b4e7d3ac75cf.msu to output.
VERBOSE: Checking Microsoft Update Catalog for Id: afadc5b5-d984-4ad2-908c-75eb57be1422.
VERBOSE: POST http://www.catalog.update.microsoft.com/DownloadDialog.aspx with -1-byte payload
VERBOSE: received 32102-byte response of content type text/html; charset=utf-8
VERBOSE: Adding http://download.windowsupdate.com/c/msdownload/update/software/updt/2018/10/windows10.0-kb4462930-x86_cc8938fa7fd394c8948d1b5dfc84de11715edb8a.msu to output.
VERBOSE: Checking Microsoft Update Catalog for Id: c9592c09-3f5f-4e82-89ce-63ab15bdb770.
VERBOSE: POST http://www.catalog.update.microsoft.com/DownloadDialog.aspx with -1-byte payload
VERBOSE: received 32102-byte response of content type text/html; charset=utf-8
VERBOSE: Adding http://download.windowsupdate.com/d/msdownload/update/software/updt/2018/10/windows10.0-kb4462930-x86_380e274de951748a3276fe3db5e864e878737111.msu to output.
VERBOSE: Checking Microsoft Update Catalog for Id: 71af9579-b9ba-433a-8eb6-62be02512547.
VERBOSE: POST http://www.catalog.update.microsoft.com/DownloadDialog.aspx with -1-byte payload
VERBOSE: received 32172-byte response of content type text/html; charset=utf-8
VERBOSE: Adding http://download.windowsupdate.com/d/msdownload/update/software/updt/2018/10/windows10.0-kb4462930-x64_aa56f192a75aac0029cd9a39d5ebf3eb625feebd.msu to output.
VERBOSE: Checking Microsoft Update Catalog for Id: 5808878a-e039-45be-9efa-626fe6c768ee.
VERBOSE: POST http://www.catalog.update.microsoft.com/DownloadDialog.aspx with -1-byte payload
VERBOSE: received 32172-byte response of content type text/html; charset=utf-8
VERBOSE: Adding http://download.windowsupdate.com/c/msdownload/update/software/updt/2018/10/windows10.0-kb4462930-x64_283d641ac17b77160816aa7dd16da317df6bc72e.msu to output.
VERBOSE: Checking Microsoft Update Catalog for Id: 3746fcb6-6fce-44f1-9ecc-5d88004e80fb.
VERBOSE: POST http://www.catalog.update.microsoft.com/DownloadDialog.aspx with -1-byte payload
VERBOSE: received 32181-byte response of content type text/html; charset=utf-8
VERBOSE: Adding http://download.windowsupdate.com/d/msdownload/update/software/updt/2018/10/windows10.0-kb4462930-x64_b8f616bfe68d36da56bb5451256ac8b6fb99d986.msu to output.
VERBOSE: Checking Microsoft Update Catalog for Id: 6ed42ebf-9bd1-436c-9d3d-390c04aa0a58.
VERBOSE: POST http://www.catalog.update.microsoft.com/DownloadDialog.aspx with -1-byte payload
VERBOSE: received 32172-byte response of content type text/html; charset=utf-8
VERBOSE: Adding http://download.windowsupdate.com/c/msdownload/update/software/updt/2018/10/windows10.0-kb4462930-x64_07407891b70292a936ebf6f115c2b0a410b44654.msu to output.
VERBOSE: Checking Microsoft Update Catalog for Id: 32c3228d-6964-40e2-ba8a-d82487b11df0.
VERBOSE: POST http://www.catalog.update.microsoft.com/DownloadDialog.aspx with -1-byte payload
VERBOSE: received 32208-byte response of content type text/html; charset=utf-8
VERBOSE: Adding http://download.windowsupdate.com/d/msdownload/update/software/updt/2018/10/windows8-rt-kb4462930-x64_8eae6587ff9d81d111d9db2a86f6a107cc5c576e.msu to output.
VERBOSE: Checking Microsoft Update Catalog for Id: 7d740e57-8469-46c9-8392-9f365d5a7841.
VERBOSE: POST http://www.catalog.update.microsoft.com/DownloadDialog.aspx with -1-byte payload
VERBOSE: received 32142-byte response of content type text/html; charset=utf-8
VERBOSE: Adding http://download.windowsupdate.com/d/msdownload/update/software/updt/2018/10/windows8-rt-kb4462930-x86_a6d56751867ed0468280475125f4e4d0ea2a76da.msu to output.
VERBOSE: Checking Microsoft Update Catalog for Id: 7aba6a05-fb0e-4002-a1d1-eb272c2df7c3.
VERBOSE: POST http://www.catalog.update.microsoft.com/DownloadDialog.aspx with -1-byte payload
VERBOSE: received 32112-byte response of content type text/html; charset=utf-8
VERBOSE: Adding http://download.windowsupdate.com/d/msdownload/update/software/updt/2018/10/windows8.1-kb4462930-x64_bbaadf6e2537a54b7b277ab8349dfd1b7aa354c8.msu to output.
VERBOSE: Checking Microsoft Update Catalog for Id: 4240aaf5-b99a-462f-95d6-76b3088735e1.
VERBOSE: POST http://www.catalog.update.microsoft.com/DownloadDialog.aspx with -1-byte payload
VERBOSE: received 32105-byte response of content type text/html; charset=utf-8
VERBOSE: Adding http://download.windowsupdate.com/d/msdownload/update/software/updt/2018/10/windows8-rt-kb4462930-x64_8eae6587ff9d81d111d9db2a86f6a107cc5c576e.msu to output.
VERBOSE: Checking Microsoft Update Catalog for Id: 29fad9f7-bf82-413a-90c6-4537d3141789.
VERBOSE: POST http://www.catalog.update.microsoft.com/DownloadDialog.aspx with -1-byte payload
VERBOSE: received 32067-byte response of content type text/html; charset=utf-8
VERBOSE: Adding http://download.windowsupdate.com/d/msdownload/update/software/updt/2018/10/windows8.1-kb4462930-x86_7762ebdf20fb0b1fb64cbc92ddba4582bc3fac0a.msu to output.
VERBOSE: Checking Microsoft Update Catalog for Id: f0bd488b-05da-416c-b619-10bca39c9cd8.
VERBOSE: POST http://www.catalog.update.microsoft.com/DownloadDialog.aspx with -1-byte payload
VERBOSE: received 32131-byte response of content type text/html; charset=utf-8
VERBOSE: Adding http://download.windowsupdate.com/d/msdownload/update/software/updt/2018/10/windows8.1-kb4462930-x64_bbaadf6e2537a54b7b277ab8349dfd1b7aa354c8.msu to output.


KB      : KB4462930
Arch    : x86
Version : 8Embedded
Note    : Update for Adobe Flash Player for Windows Embedded 8 standard (KB4462930)
URL     : http://download.windowsupdate.com/d/msdownload/update/software/updt/2018/10/windows8-rt-kb4462930-x86_a6d56751867ed0468280475125f4e4d0ea2a76da.msu

KB      : KB4462930
Arch    : x64
Version : 8Embedded
Note    : Update for Adobe Flash Player for Windows Embedded 8 standard for x64-based Systems (KB4462930)
URL     : http://download.windowsupdate.com/d/msdownload/update/software/updt/2018/10/windows8-rt-kb4462930-x64_8eae6587ff9d81d111d9db2a86f6a107cc5c576e.msu

KB      : KB4462930
Arch    : x64
Version : 8.1
Note    : Update for Adobe Flash Player for Windows 8.1 for x64-based Systems (KB4462930)
URL     : http://download.windowsupdate.com/d/msdownload/update/software/updt/2018/10/windows8.1-kb4462930-x64_bbaadf6e2537a54b7b277ab8349dfd1b7aa354c8.msu

KB      : KB4462930
Arch    : x86
Version : 8.1
Note    : Update for Adobe Flash Player for Windows 8.1 (KB4462930)
URL     : http://download.windowsupdate.com/d/msdownload/update/software/updt/2018/10/windows8.1-kb4462930-x86_7762ebdf20fb0b1fb64cbc92ddba4582bc3fac0a.msu

KB      : KB4462930
Arch    : x64
Version : 2012R2
Note    : Update for Adobe Flash Player for Windows Server 2012 R2 (KB4462930)
URL     : http://download.windowsupdate.com/d/msdownload/update/software/updt/2018/10/windows8.1-kb4462930-x64_bbaadf6e2537a54b7b277ab8349dfd1b7aa354c8.msu

KB      : KB4462930
Arch    : x64
Version : 2012
Note    : Update for Adobe Flash Player for Windows Server 2012 (KB4462930)
URL     : http://download.windowsupdate.com/d/msdownload/update/software/updt/2018/10/windows8-rt-kb4462930-x64_8eae6587ff9d81d111d9db2a86f6a107cc5c576e.msu

KB      : KB4462930
Arch    : ARM64
Version : 1809
Note    : Update for Adobe Flash Player for Windows 10 Version 1809 for ARM64-based Systems (KB4462930)
URL     : http://download.windowsupdate.com/d/msdownload/update/software/updt/2018/10/windows10.0-kb4462930-arm64_4331c3089be4dfd515059818f8b6c015c5643ad0.msu

KB      : KB4462930
Arch    : x86
Version : 1809
Note    : Update for Adobe Flash Player for Windows 10 Version 1809 (KB4462930)
URL     : http://download.windowsupdate.com/d/msdownload/update/software/updt/2018/10/windows10.0-kb4462930-x86_ee44cbbd86b3271990ccc472760f0228094131e5.msu

KB      : KB4462930
Arch    : x64
Version : 1809
Note    : Update for Adobe Flash Player for Windows 10 Version 1809 for x64-based Systems (KB4462930)
URL     : http://download.windowsupdate.com/d/msdownload/update/software/updt/2018/10/windows10.0-kb4462930-x64_b22236a150c0fffaa1b132c7dfbbc844ad6c45e3.msu

KB      : KB4462930
Arch    : x64
Version : 1809
Note    : Update for Adobe Flash Player for Windows Server 2019 (1809) for x64-based Systems (KB4462930)
URL     : http://download.windowsupdate.com/d/msdownload/update/software/updt/2018/10/windows10.0-kb4462930-x64_b22236a150c0fffaa1b132c7dfbbc844ad6c45e3.msu

KB      : KB4462930
Arch    : ARM64
Version : 1803
Note    : Update for Adobe Flash Player for Windows 10 Version 1803 for ARM64-based Systems (KB4462930)
URL     : http://download.windowsupdate.com/c/msdownload/update/software/updt/2018/10/windows10.0-kb4462930-arm64_24b09da230c6a260882df62d211e6d1a26ad5dc0.msu

KB      : KB4462930
Arch    : x86
Version : 1803
Note    : Update for Adobe Flash Player for Windows 10 Version 1803 (KB4462930)
URL     : http://download.windowsupdate.com/c/msdownload/update/software/updt/2018/10/windows10.0-kb4462930-x86_b1bd42a524bb4abbd08f66090b69f9432a756de2.msu

KB      : KB4462930
Arch    : x64
Version : 1803
Note    : Update for Adobe Flash Player for Windows 10 Version 1803 for x64-based Systems (KB4462930)
URL     : http://download.windowsupdate.com/c/msdownload/update/software/updt/2018/10/windows10.0-kb4462930-x64_3a4a1cada2d518585873b6cfdad8b4e7d3ac75cf.msu

KB      : KB4462930
Arch    : x64
Version : 1709
Note    : Update for Adobe Flash Player for Windows 10 Version 1709 for x64-based Systems (KB4462930)
URL     : http://download.windowsupdate.com/c/msdownload/update/software/updt/2018/10/windows10.0-kb4462930-x64_07407891b70292a936ebf6f115c2b0a410b44654.msu

KB      : KB4462930
Arch    : x86
Version : 1709
Note    : Update for Adobe Flash Player for Windows 10 Version 1709 (KB4462930)
URL     : http://download.windowsupdate.com/c/msdownload/update/software/updt/2018/10/windows10.0-kb4462930-x86_cc8938fa7fd394c8948d1b5dfc84de11715edb8a.msu

KB      : KB4462930
Arch    : ARM64
Version : 1709
Note    : Update for Adobe Flash Player for Windows 10 Version 1709  for ARM64-based Systems (KB4462930)
URL     : http://download.windowsupdate.com/c/msdownload/update/software/updt/2018/10/windows10.0-kb4462930-arm64_332ecfb5958a87eb6e37206cbd9cf1aa2339659e.msu

KB      : KB4462930
Arch    : x86
Version : 1703
Note    : Update for Adobe Flash Player for Windows 10 Version 1703 (KB4462930)
URL     : http://download.windowsupdate.com/c/msdownload/update/software/updt/2018/10/windows10.0-kb4462930-x86_f3fb8219e526d228e9ae9b44973d7d73376f3367.msu

KB      : KB4462930
Arch    : x64
Version : 1703
Note    : Update for Adobe Flash Player for Windows 10 Version 1703 for x64-based Systems (KB4462930)
URL     : http://download.windowsupdate.com/c/msdownload/update/software/updt/2018/10/windows10.0-kb4462930-x64_283d641ac17b77160816aa7dd16da317df6bc72e.msu

KB      : KB4462930
Arch    : x64
Version : 1607
Note    : Update for Adobe Flash Player for Windows 10 Version 1607 for x64-based Systems (KB4462930)
URL     : http://download.windowsupdate.com/d/msdownload/update/software/updt/2018/10/windows10.0-kb4462930-x64_aa56f192a75aac0029cd9a39d5ebf3eb625feebd.msu

KB      : KB4462930
Arch    : x86
Version : 1607
Note    : Update for Adobe Flash Player for Windows 10 Version 1607 (KB4462930)
URL     : http://download.windowsupdate.com/d/msdownload/update/software/updt/2018/10/windows10.0-kb4462930-x86_380e274de951748a3276fe3db5e864e878737111.msu

KB      : KB4462930
Arch    : x64
Version : 1607
Note    : Update for Adobe Flash Player for Windows Server 2016 for x64-based Systems (KB4462930)
URL     : http://download.windowsupdate.com/d/msdownload/update/software/updt/2018/10/windows10.0-kb4462930-x64_aa56f192a75aac0029cd9a39d5ebf3eb625feebd.msu

KB      : KB4462930
Arch    : x64
Version : 1507
Note    : Update for Adobe Flash Player for Windows 10 Version 1507 for x64-based Systems (KB4462930)
URL     : http://download.windowsupdate.com/d/msdownload/update/software/updt/2018/10/windows10.0-kb4462930-x64_b8f616bfe68d36da56bb5451256ac8b6fb99d986.msu

KB      : KB4462930
Arch    : x86
Version : 1507
Note    : Update for Adobe Flash Player for Windows 10 Version 1507 (KB4462930)
URL     : http://download.windowsupdate.com/d/msdownload/update/software/updt/2018/10/windows10.0-kb4462930-x86_72c899f691403365053da2c53d1097f5065da3c7.msu
```

**_If it is not a bug, please remove the rest of this template._**

## Steps to Reproduce

Please provide detailed steps for reproducing the issue.

1. Run Get-LatestFlash

## Context

Please provide any relevant information about your setup. This is important in case the issue is not reproducible except for under certain conditions.

* **LatestUpdate Version**:  2.4.0.89
* **PowerShell Version**: 5.1 Build 14409 Revision 1018
* **Operating System**: Server 2012 R2

## Failure Logs

Please include any relevant log snippets or files here.

**Comment by f0nt4** [comment]
I also ran into the issue that the update for Windows Server 2019 was not in the list. From what I found is that the catalog website only returns 25 items per page. The Get-UpdateCatalogLink does a webrequest to the search url and therefor only get 25 results. In the meanwhile, the results on the next page(s) don't get retrieved.
There're a couple of solutions I guess.
- Detect the "Next" button on the page and also get those results.
- Define a OS name so the search will be more specific
- Don't scrape the page but find a way to query the API, which would be best I guess.
